src/irmin-pack/IO.ml

```diff
@@ -232,25 +259,27 @@ module Unix : S = struct
     mkdir (Filename.dirname file);
     match Sys.file_exists file with
     | false ->
-        let x = Unix.openfile file Unix.[ O_CREAT; mode; O_CLOEXEC ] 0o644 in
-        let raw = Raw.v x in
-        Raw.unsafe_set_offset raw 0L;
-        Raw.unsafe_set_version raw current_version;
-        v ~offset:0L ~version:current_version raw
+        if readonly then v ~offset:0L ~version:current_version None
+        else
+          let x = Unix.openfile file Unix.[ O_CREAT; mode; O_CLOEXEC ] 0o644 in
+          let raw = Raw.v x in
+          Raw.unsafe_set_offset raw 0L;
+          Raw.unsafe_set_version raw current_version;
+          v ~offset:0L ~version:current_version (Some raw)
     | true ->
         let x = Unix.openfile file Unix.[ O_EXCL; mode; O_CLOEXEC ] 0o644 in
```

Comment:
As mentioned on Slack, I think this `O_EXCL` may be an existing bug.

Reply:
I've removed `O_EXCL` when the file exists and add it to when the file does not exists so that it can be used with `O_CREAT`. 
It seems that it is also a bug in [index_unix](https://github.com/mirage/index/blob/master/src/unix/index_unix.ml#L305)